### PR TITLE
Makes not using VSCode an error

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

It needs to be an error, they can't just choose to compile it in DM anyways and expect it to work.
Also removed mention of `BUILD.bat` to encourage VSCode adoption. They need the linter to yell at them.

## Changelog
:cl:
code: Made it more explicit that coders need to be compiling with VSCode, not BYOND's Dream Maker editor.
tweak: Adjusted the error message to also mention installing the recommended VSCode extensions.
/:cl:

### CODE REVIEW

File: code/_compile_options.dm [hunk 1]
```diff
@@ -109,9 +109,8 @@
 #endif
 
 #if !defined(CBT) && !defined(SPACEMAN_DMM)
-#warn Building with Dream Maker is no longer supported and will result in errors.
-#warn In order to build, run BUILD.bat in the root directory.
-#warn Consider switching to VSCode editor instead, where you can press Ctrl+Shift+B to build.
+#error Building with Dream Maker is no longer supported and will result in errors.
+#error Switch to the VSCode text editor instead, where you can press Ctrl+Shift+B to build.
```
Comment (ike709) [inline code/_compile_options.dm]:
```suggestion
#error Switch to the VSCode text editor instead, where you can press Ctrl+Shift+B to build (or VSCode's other build buttons).
```

File: code/_compile_options.dm [hunk 2]
```diff
@@ -109,9 +109,8 @@
 #endif
 
 #if !defined(CBT) && !defined(SPACEMAN_DMM)
-#warn Building with Dream Maker is no longer supported and will result in errors.
-#warn In order to build, run BUILD.bat in the root directory.
-#warn Consider switching to VSCode editor instead, where you can press Ctrl+Shift+B to build.
+#error Building with Dream Maker is no longer supported and will result in errors.
+#error Switch to the VSCode text editor instead, where you can press Ctrl+Shift+B to build (or VSCode's other build buttons).
```
Comment (ike709) [inline code/_compile_options.dm]:
```suggestion
#error Switch to the VSCode text editor instead, where you can press Ctrl+Shift+B to build (or VSCode's other build buttons).
#error And don't forget to add the recommended VSCode extensions. You'll be prompted when you first open the codebase in VSCode.
```

File: code/_compile_options.dm [hunk 3]
```diff
@@ -109,9 +109,9 @@
 #endif
 
 #if !defined(CBT) && !defined(SPACEMAN_DMM)
-#warn Building with Dream Maker is no longer supported and will result in errors.
-#warn In order to build, run BUILD.bat in the root directory.
```
Comment (Archanial) [inline code/_compile_options.dm:113]: Makes this an error message too, people should know this exists

Comment (Crossedfall) [reply code/_compile_options.dm:113]: If they're at the point of using the scripts directly, they probably don't need an error telling them about it. For the general user, I believe the intent is to push them entirely towards vscode.

File: code/_compile_options.dm [hunk 4]
```diff
@@ -109,9 +109,9 @@
...
-#warn Consider switching to VSCode editor instead, where you can press Ctrl+Shift+B to build.
+#error Building with Dream Maker is no longer supported and will result in errors.
+#error Switch to the VSCode text editor instead, where you can press Ctrl+Shift+B to build (or VSCode's other build buttons).
```
Comment (Crossedfall) [inline code/_compile_options.dm]:
```suggestion
#error Switch to VSCode and when prompted install the recommended extensions, you can then either use the UI or press Ctrl+Shift+B to build the codebase.
```

File: code/_compile_options.dm [hunk 5]
```diff
@@ -109,9 +109,9 @@
...
+#error And don't forget to add the recommended VSCode extensions. You'll be prompted when you first open the codebase in VSCode.
```
Comment (Crossedfall) [inline code/_compile_options.dm]: This is super clunky. Better to just slightly rework the prior sentence to mention the extensions.
```suggestion
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]

**Review by Crossedfall — APPROVED** [review]

**Comment by Crossedfall** [comment]
Don't really know if this is needed, but, meh.

**Pull request merged**